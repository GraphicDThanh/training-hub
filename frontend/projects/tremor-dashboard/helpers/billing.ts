// Helpers
import { moneyFormat } from "@/helpers";

// Constants
import { TRANSACTION_CLASS } from "@/constants";

// Types
import { STATUS_LIST } from "@/types";

export const getContentByProps = (
  status: number,
  type: number,
  amount: number,
) => {
  const formattedAmount = moneyFormat({ value: amount });
  const hasStatus = status !== STATUS_LIST.ERROR;
  const hasType = type !== STATUS_LIST.ERROR;

  if (!hasStatus) {
    return {
      ...TRANSACTION_CLASS.PENDING,
      value: "Pending",
    };
  }

  if (hasType) {
    return {
      ...TRANSACTION_CLASS.DECREASE,
      value: `-$${formattedAmount}`,
    };
  }

  return {
    ...TRANSACTION_CLASS.INCREASE,
    value: `+$${formattedAmount}`,
  };
};
