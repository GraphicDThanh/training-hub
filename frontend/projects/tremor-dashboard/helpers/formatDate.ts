import { FORMAT_DATE } from "@/constants";
import dayjs from "dayjs";
import { convertDateTimeToObject, convertDateToDateTime } from "./time";

const padTo2Digits = (num: number) => {
  return num.toString().padStart(2, "0");
};

export const formatDate = (date: Date) => {
  return [
    padTo2Digits(date.getMonth() + 1),
    padTo2Digits(date.getDate()),
    date.toLocaleDateString("en", { year: "2-digit" }),
  ].join(".");
};

/**
 * @param value string - Date time string to format
 * @returns string - Formatted date time
 */
export const formatDateTime = (value: string) => {
  return dayjs(value).format("MMM D, YYYY h:mm A");
};

/**
 * @param value string - Date string to format
 * @returns string - Formatted date
 */
export const formatNewDate = (value: string) => {
  return dayjs(value).format("MMM D, YYYY");
};

export const covertDate = (value: any) => {
  const currentDate = convertDateTimeToObject(value);
  return convertDateToDateTime(currentDate, FORMAT_DATE.YEAR_MONTH_DATE_TIME);
};
