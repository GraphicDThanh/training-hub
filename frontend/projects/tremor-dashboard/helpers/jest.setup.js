jest.mock("next/dynamic", () => ({
  __esModule: true,
  default: (...props) => {
    const dynamicModule = jest.requireActual("next/dynamic");
    const dynamicActualComp = dynamicModule.default;
    const RequiredComponent = dynamicActualComp(props[0]);
    RequiredComponent.preload
      ? RequiredComponent.preload()
      : RequiredComponent.render.preload();
    return RequiredComponent;
  },
}));

jest.mock("next/navigation", () => {
  const actual = jest.mock("next/navigation");
  return {
    ...actual,
    useRouter: jest.fn(() => ({
      push: jest.fn(),
      replace: jest.fn(),
    })),
    useSearchParams: jest.fn(() => ({ get: jest.fn(() => "mockPage") })),
    usePathname: jest.fn(),
  };
});

jest.mock("next/cache", () => ({
  revalidatePath: jest.fn(),
  revalidateTag: jest.fn(),
}));
