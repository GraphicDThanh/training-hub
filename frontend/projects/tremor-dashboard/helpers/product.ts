import { Product } from "@/types";

export const getOtherProducts = (
  products: Product[],
  id: number,
  length: number,
) => {
  const productsLength = products.length;
  const otherProducts: Product[] = [];

  for (let i = 0; i < productsLength; i++) {
    const product = products[i];
    if (product.id.toString() !== id.toString()) otherProducts.push(product);
    if (otherProducts.length === length) break;
  }

  return otherProducts;
};
