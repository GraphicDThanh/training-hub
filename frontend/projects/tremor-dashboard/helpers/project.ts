import { ImageProject, Project, User } from "@/types";
import { getSubAttrsByKey } from "./common";

export const convertProjectData = (
  data: Project,
  projectTools: ImageProject[],
  projectParticipants: User[],
) => {
  const { name, description, dueDate, participants, tools } = data;

  const participantsValue = getSubAttrsByKey(
    projectParticipants,
    participants,
    "id",
    "email",
  );
  const toolsValue = getSubAttrsByKey(projectTools, tools, "id", "name");

  return {
    name,
    description,
    tools: toolsValue,
    participants: participantsValue,
    dueDate: new Date(dueDate),
  };
};
