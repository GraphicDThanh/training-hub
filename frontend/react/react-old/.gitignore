node_modules
out
dist
.DS_Store
.git
.docpad.db
vendor/
.vagrant
bower-components/
bower_components/
npm-debug.log
.gitignore~
vagrant/Vagrantfile~
build/
