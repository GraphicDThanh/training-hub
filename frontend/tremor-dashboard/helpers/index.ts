export * from "./formatDate";
export * from "./formatFormData";
export * from "./common";
export * from "./format";
export * from "./formatNumber";
export * from "./billing";
export * from "./apis";
export * from "./pincode";
export * from "./time";
export * from "./user";
export * from "./project";
export * from "./validate";
export * from "./colorScheme";
export * from "./styles";
