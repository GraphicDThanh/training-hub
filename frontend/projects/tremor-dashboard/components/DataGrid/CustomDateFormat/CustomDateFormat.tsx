"use client";

// Components
import { Text } from "@tremor/react";

// Helpers
import { formatDateTime } from "@/helpers/formatDate";

const CustomDateFormat = ({ date }: { date: string }) => (
  <Text className="text-xs text-tertiary dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] w-[180px] truncate">
    {formatDateTime(date)}
  </Text>
);

export default CustomDateFormat;
