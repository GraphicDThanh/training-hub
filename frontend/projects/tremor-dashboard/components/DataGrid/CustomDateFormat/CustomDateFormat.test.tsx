import { render } from "@testing-library/react";

// Helpers
import { formatDateTime } from "@/helpers/formatDate";

// Components
import CustomDateFormat from "./CustomDateFormat";

const CustomDateFormatComponent = () =>
  render(<CustomDateFormat date="01 Nov, 11:20 AM" />);

const CustomDateFormatEmpty = () => render(<CustomDateFormat date="" />);

describe("CustomDateFormat component", () => {
  it("should render snapshot correctly", () => {
    const { container } = CustomDateFormatComponent();

    expect(container).toMatchSnapshot();
  });

  it("should render date with formatDateTime correct", () => {
    const { getByText } = CustomDateFormatComponent();
    const dateText = getByText(formatDateTime("01 Nov, 11:20 AM"));

    expect(dateText).toBeTruthy();
  });

  it("should show error if empty date", () => {
    const { getByText } = CustomDateFormatEmpty();
    const dateText = getByText("Invalid Date");

    expect(dateText).toBeTruthy();
  });
});
