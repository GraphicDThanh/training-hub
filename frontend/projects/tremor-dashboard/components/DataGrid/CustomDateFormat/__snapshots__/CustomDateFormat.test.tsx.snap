// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDateFormat component should render snapshot correctly 1`] = `
<div>
  <p
    class="text-tremor-default text-xs text-tertiary dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] w-[180px] truncate"
  >
    Nov 1, 2001 11:20 AM
  </p>
</div>
`;
